rancilio-pid/rancilio-pid.ino

```diff
@@ -1100,9 +1261,21 @@ void updateState() {
         }
       }
 
+      /* STATE 6 (Steam) DETECTION */
+
+      if (Input > setPoint + steamZoneTemperatureDifference && activeState != 6) {
```

Comment:
Ich versteh nicht den Sinn dieser "Steam Detection":
- activeState=6 wurde bereits verlaesslich direkt über generateSteam() gesetzt.
- Input kann sonst nur > SetPoint+steamZoneTemperatureDifference  sein, 
- a) wenn die PID Steuerung (im State=3) deutlich übersteuert hat (sollte SEEEHR selten vorkommen und wenn ja, dann darf das Steam dann auch nicht einfach anschalten)  ODER
 - b) der Hardware-Steam-Knopf gedrückt wurde und dieser die Heizung ohne PID auf 100% hält. In  diesem Fall finde ich es gefährlich, wenn dem Anwender im Display der Eindruck erweckt wird als würde die PID auf den SteamSetPoint steuern.

Daher würde ich diesen Block eher (derzeit) entfernen. Spricht was dagegen?

Reply:
Also ich hatte das Problem, das bei mir Tv größer 0 dazu geführt hat, das die Maschine beim abkühlen krass nachgeheizt hat weil die negative Temperaturänderung mal Faktor Kd zu einem hohen Output führt. Hat meiner Meinung nach ein ungewünschtes Verhalten, besonders beim abkühlen.
Damit jetzt auch bei einem Bedienfehler, Maschine aus versehen ausgeschaltet oder beim Dampferzeugen mit dem Dampfthermostat nicht auch unnötigerweise output beim abkühlen erzeugt wird habe ich die den Teil eingesetzt.
Die Idee hinter der Anzeige des setpoints kann ich verstehen. für mich ist es jedoch von Vorteil wenn ich nach dem Dampf erzeugen sehe das der setpoint wieder bei 94°C ist.

Reply:
Verstehe.. Das leuchtet ein (1. er sollte nicht mehr heizen 2. und der brewSetPoint sollte angezeigt werden).  Ich hab das mal ein wenig umgebaut. Wäre gut, wenn du später das Verhalten testest um zu sehen ob das so wie von dir gewünscht funktioniert.